mmv1/products/vertexai/ReasoningEngine.yaml

```diff
@@ -0,0 +1,202 @@
+# Copyright 2025 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'ReasoningEngine'
+description: |-
+  ReasoningEngine provides a customizable runtime for models to determine which actions to take and in which order.
+references:
+  guides:
+    'Develop and deploy agents on Vertex AI Agent Engine': 'https://cloud.google.com/vertex-ai/generative-ai/docs/agent-engine/quickstart'
+  api: 'https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.reasoningEngines/'
+docs:
+base_url: '{{name}}'
+self_link: '{{name}}'
+create_url: 'projects/{{project}}/locations/{{region}}/reasoningEngines'
+update_verb: 'PATCH'
+update_mask: true
+timeouts:
+  insert_minutes: 20
+  update_minutes: 20
+  delete_minutes: 20
+async:
+  actions: ['create', 'delete', 'update']
+  type: 'OpAsync'
+  operation:
+    base_url: '{{op_id}}'
+  result:
+    resource_inside_response: true
+custom_code:
+  custom_import: 'templates/terraform/custom_import/self_link_as_name.tmpl'
+examples:
+  - name: 'vertex_ai_reasoning_engine_basic'
+    primary_resource_id: 'reasoning_engine'
+    exclude_docs: true
+    vars:
+      name: 'reasoning-engine'
+  - name: 'vertex_ai_reasoning_engine_full'
+    primary_resource_id: 'reasoning_engine'
+    vars:
+      name: 'reasoning-engine'
+      bucket_name: 'reasoning-engine'
+      kms_key_name: 'example-key'
+      secret_name: 'secret'
+      service_account_id: 'sa'
+    test_vars_overrides:
+      'kms_key_name': 'acctest.BootstrapKMSKeyWithPurposeInLocationAndName(t, "ENCRYPT_DECRYPT", "us-central1", "tf-bootstrap-re-key1").CryptoKey.Name'
+    external_providers: ["time"]
+parameters:
+  - name: 'region'
+    type: String
+    description: The region of the reasoning engine. eg us-central1
+    url_param_only: true
+    immutable: true
+properties:
+  - name: 'name'
+    type: String
+    description: |-
+      The generated name of the ReasoningEngine, in the format
+      'projects/{project}/locations/{location}/reasoningEngines/{reasoningEngine}'
+    output: true
+  - name: 'displayName'
+    type: String
+    description: The display name of the ReasoningEngine.
+    required: true
+  - name: 'description'
+    type: String
+    description: The description of the ReasoningEngine.
+  - name: 'etag'
```

Comment:
IMO just remove this unless it's a `type: Fingerprint`. If it isn't required, no need to include at all for now.